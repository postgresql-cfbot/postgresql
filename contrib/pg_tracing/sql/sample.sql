-- Trace nothing
SET pg_tracing.sample_rate = 0.0;
SET pg_tracing.caller_sample_rate = 0.0;

-- Query with sampling flag
/*dddbs='postgres.db',traceparent='00-00000000000000000000000000000021-0000000000000021-01'*/ SELECT 1;
-- Query without trace context
SELECT 1;

-- No spans should have been generated
select count(distinct(trace_id)) from pg_tracing_consume_spans;

-- Enable full sampling
SET pg_tracing.sample_rate = 1.0;

-- Generate queries with sampling flag on, off and no trace context
/*dddbs='postgres.db',traceparent='00-00000000000000000000000000000022-0000000000000022-01'*/ SELECT 1;
/*dddbs='postgres.db',traceparent='00-00000000000000000000000000000023-0000000000000023-00'*/ SELECT 2;
SELECT 3;
SELECT 4;

-- Check number of generated spans
select count(distinct(trace_id)) from pg_tracing_peek_spans;
-- Check span order
select span_operation, parameters from pg_tracing_peek_spans order by span_start, span_start_ns;
-- Top spans should reuse generated ids and have trace_id = parent_id
select span_operation, parameters from pg_tracing_consume_spans where trace_id = parent_id order by span_start, span_start_ns;

-- Only trace query with sampled flag
SET pg_tracing.sample_rate = 0.0;
SET pg_tracing.caller_sample_rate = 1.0;

-- Generate queries with sampling flag on, off and no trace context
/*dddbs='postgres.db',traceparent='00-00000000000000000000000000000024-0000000000000024-01'*/ SELECT 1;
/*dddbs='postgres.db',traceparent='00-00000000000000000000000000000025-0000000000000025-00'*/ SELECT 2;
SELECT 1;

-- Check number of generated spans
select count(distinct(trace_id)) from pg_tracing_peek_spans;
-- Check span ordering
select span_operation, parameters from pg_tracing_consume_spans order by span_start, span_start_ns;
