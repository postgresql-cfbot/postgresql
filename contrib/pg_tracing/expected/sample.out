-- Trace nothing
SET pg_tracing.sample_rate = 0.0;
SET pg_tracing.caller_sample_rate = 0.0;
-- Query with sampling flag
/*dddbs='postgres.db',traceparent='00-00000000000000000000000000000021-0000000000000021-01'*/ SELECT 1;
 ?column? 
----------
        1
(1 row)

-- Query without trace context
SELECT 1;
 ?column? 
----------
        1
(1 row)

-- No spans should have been generated
select count(distinct(trace_id)) from pg_tracing_peek_spans;
 count 
-------
     0
(1 row)

-- Enable full sampling
SET pg_tracing.sample_rate = 1.0;
-- Generate queries with sampling flag on, off and no trace context
/*dddbs='postgres.db',traceparent='00-00000000000000000000000000000022-0000000000000022-01'*/ SELECT 1;
 ?column? 
----------
        1
(1 row)

/*dddbs='postgres.db',traceparent='00-00000000000000000000000000000023-0000000000000023-00'*/ SELECT 2;
 ?column? 
----------
        2
(1 row)

SELECT 3;
 ?column? 
----------
        3
(1 row)

SELECT 4;
 ?column? 
----------
        4
(1 row)

-- Check number of generated spans
select count(distinct(trace_id)) from pg_tracing_peek_spans;
 count 
-------
     4
(1 row)

-- Check span order
select resource, parameters from pg_tracing_peek_spans order by span_start, span_start_ns;
  resource  | parameters 
------------+------------
 SELECT $1; | $1 = 1
 Parse      | 
 Post Parse | 
 Planner    | 
 Start      | 
 Run        | 
 Result     | 
 Finish     | 
 End        | 
 SELECT $1; | $1 = 2
 Parse      | 
 Post Parse | 
 Planner    | 
 Start      | 
 Run        | 
 Result     | 
 Finish     | 
 End        | 
 SELECT $1; | $1 = 3
 Parse      | 
 Post Parse | 
 Planner    | 
 Start      | 
 Run        | 
 Result     | 
 Finish     | 
 End        | 
 Parse      | 
 SELECT $1; | $1 = 4
 Post Parse | 
 Planner    | 
 Start      | 
 Run        | 
 Result     | 
 Finish     | 
 End        | 
(36 rows)

-- Top spans should reuse generated ids and have trace_id = parent_id = span_id
select resource, parameters from pg_tracing_consume_spans where trace_id = parent_id and parent_id = span_id order by span_start, span_start_ns;
  resource  | parameters 
------------+------------
 SELECT $1; | $1 = 3
 SELECT $1; | $1 = 4
(2 rows)

-- Only trace query with sampled flag
SET pg_tracing.sample_rate = 0.0;
SET pg_tracing.caller_sample_rate = 1.0;
-- Generate queries with sampling flag on, off and no trace context
/*dddbs='postgres.db',traceparent='00-00000000000000000000000000000024-0000000000000024-01'*/ SELECT 1;
 ?column? 
----------
        1
(1 row)

/*dddbs='postgres.db',traceparent='00-00000000000000000000000000000025-0000000000000025-00'*/ SELECT 2;
 ?column? 
----------
        2
(1 row)

SELECT 1;
 ?column? 
----------
        1
(1 row)

-- Check number of generated spans
select count(distinct(trace_id)) from pg_tracing_peek_spans;
 count 
-------
     2
(1 row)

-- Check span ordering
select resource, parameters from pg_tracing_consume_spans order by span_start, span_start_ns;
             resource              | parameters 
-----------------------------------+------------
 SET pg_tracing.sample_rate = 0.0; | 
 Parse                             | 
 Post Parse                        | 
 SET pg_tracing.sample_rate = 0.0; | 
 SELECT $1;                        | $1 = 1
 Parse                             | 
 Post Parse                        | 
 Planner                           | 
 Start                             | 
 Run                               | 
 Result                            | 
 Finish                            | 
 End                               | 
(13 rows)

